### DISCUSSION TITLE
Domoticz receiving auto discovery events multiple times

### DISCUSSION THREAD

**Discussion by BuzzDoet** [zwave-js/zwave-js-ui#3907] [General]
Hi,
I'm a newby regarding zwave-je-ui. Up and till now I used OpenZwave in Domotics, but with an update of Domoticz it became clear to me that I have to use zwave-js-ui instead.
I followed the instructions on the [Domoticz wiki ](https://www.domoticz.com/wiki/Zwave-JS-UI).
I have installed:
-Domoticz version: 2024.7 (build 16253) and 
-zwave-js-ui: 9.8.2.c09b638
-zwave-js: 12.4.3
-home id: undefined
-home hex: undefined

Domoticz, zwave-js-ui and MQTT are all running in there own Docker container.
When I look in the Devices tab in Domoticz I see all kinds of zwave devices, what I expect.
But I also see dat zwave-js-ui keeps on sending auto discovery messages, so the number of Devices in Domoticz keeps on growing and contains the same devices multiple times.

I have checked the zwave-js-ui settings but don't know which one to adjust.
The control panel in zwave-js-ui shows 18 devices which is what I expect.
Any help is welcome?
Regards, Emiel

**Comment by robertsLando** [comment]
Hi @BuzzDoet, in order to debug this I need application logs, could you share them? https://zwave-js.github.io/zwave-js-ui/#/troubleshooting/generating-logs?id=application-logs

**Reply by BuzzDoet** [reply]
Hi @robertsLando, 
Could this problem be caused by me renaming devices in Domoticz?
If so, I need to rename the devices because the name will be schown in the Domoticz UI. Can I use the Entity name template to rename device names for use in Domoticz?
If so, in which directory should I store the template and what schould its name be?
I attached the log file.
[logs.zip](https://github.com/user-attachments/files/17075240/logs.zip)

**Reply by robertsLando** [reply]
> Could this problem be caused by me renaming devices in Domoticz?

I'm not a domoticz user but I think it could be. I suggest to ask this in domoticz forum, dunno if @ccutrer can help

**Reply by ccutrer** [reply]
If you have zwave-js-ui configured to retain MQTT messages, and you rename devices in zwave-js-ui, it will send new MQTT discovery messages with the new name, and not delete the old ones (to my knowledge). If this is the cause, and since you're already in this state, the best solution would be to use a tool (like MQTT Explorer) to delete all topics under homeassistant/ that reference zwave, and the entire zwave/ tree, then restart zwave-js-ui. If it's simply Domoticz displaying duplicate devices every time zwave-js-ui sends an MQTT discovery message (i.e. if you restart zwave-js-ui, or click Rediscover Node in zwave-js-ui's control panel), that's definitely a Domoticz issue. The discovery messages are sent to the same topics each time, and also include device identifiers for the receiving software to be able to group multiple components from the same device together, but also identify that this message is for a component/entity that it has already seen.
